src/Sylius/Bundle/ApiBundle/Serializer/ShippingMethodNormalizer.php

```diff
@@ -45,24 +51,37 @@ public function normalize($object, $format = null, array $context = [])
 
         $context[self::ALREADY_CALLED] = true;
 
-        $subresourceIdentifiers = $context['subresource_identifiers'];
+        $request = $this->requestStack->getCurrentRequest();
 
-        $shipmentId = $subresourceIdentifiers['shipments'] ?? $subresourceIdentifiers['id'];
+        $filters = $request->attributes->get('_api_filters');
+        if (null === $filters) {
+            $queryString = RequestParser::getQueryString($request);
+            $filters = $queryString ? RequestParser::parseRequestParams($queryString) : null;
+        }
 
-        /** @var ShipmentInterface $shipment */
-        $shipment = $this->shipmentRepository->find($shipmentId);
+        $data = $this->normalizer->normalize($object, $format, $context);
 
-        Assert::notNull($shipment);
+        if (null === $filters) {
+            return $data;
+        }
 
-        if (isset($subresourceIdentifiers['tokenValue'])) {
-            /** @var OrderInterface|null $cart */
-            $cart = $this->orderRepository->findCartByTokenValue($subresourceIdentifiers['tokenValue']);
-            Assert::notNull($cart);
+        Assert::keyExists($filters, 'tokenValue');
+        Assert::keyExists($filters, 'shipmentId');
```

Comment:
I know, I suggested these assertions, but after second thought, can we just use early return here? The reasoning behind this is, that with these asserts we are cementing this endpoint and no one will ba able to easily add additional errors

Reply:
Do you think I should  also change `notNull` asserts on `$cart` and `$shipment`?

Reply:
It is fine, as it is